### ISSUE TITLE
Make usage of the `settings` icon more consistent and meaningful

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#62873]
### Description

Splitting this out from https://github.com/WordPress/gutenberg/pull/62129 where an interesting conversation about cohesive design language in the editor UI took place.

I believe cohesive language should apply also to icons _and their usage_ in the editor.

Some icons represent very familiar logical concepts for users. For example, a plus icon `+` immediately associates with the concepts of 'Add', 'Create new', 'Insert', or 'Increase'. 

Other icons need to be 'learned' in the context of the application and UI they are used into. This implies some initial cognitive load and learning curve. When users get familiar with the _meaning_ and _purpose_ of that icon, the icon helps identifying a specific feature in the user interface. As long as the _usage_ of that icon is consistent.

Noting that, from an accessibility perspective, proliferation of icons is a problem and it would be preferable to have icons accompanied with visible text labels, but that's another issue.

Consider the following icon:

![Screenshot 2024-06-26 at 13 32 19](https://github.com/WordPress/gutenberg/assets/1682452/f51b1a04-061a-4be2-afb4-6c32ebebe608)

Its name is `settings`. Visually, it's two sliders made of a track and a handle (or 'knob'). Regardless of the icon name `settings`, I would say the logical association isn't immediate but can be learned. It's about values that can be 'adjusted' and 'set to' so that, in a broader sense, this icon represent the concept of 'settings'.

## How this icon is actually used in the editor

At the time of writing this issue I can find 5 occurrences of `icon={ settings }` in the codebase.

In the first three cases, this icon isn't used to access 'settings'. Actually, it is used to switch a control to an alternative version where users can set custom values.

1
`SpacingInputControl` component.
By clicking the icon button the range control shows an additional input field where users can enter a custom value. There are many instances of this component for the various margin, padding, block spacing values etc.

![SpacingInputControl](https://github.com/WordPress/gutenberg/assets/1682452/59a10ee6-e581-4fc7-9690-6686fb5fa7ca)

2
`FontSizePicker` component.
Same: clicking the icon button switches the control from preset values to an alternative version where users can enter a custom value.

![FontSizePicker](https://github.com/WordPress/gutenberg/assets/1682452/a9cdd52c-05d9-4d6a-aa0c-d2497d8f7efd)

3
`ShadowInputControl` control in the `ShadowPopover` component.
Same: clicking the icon button switches the range slider to an input field where users can enter a custom value.

![ShadowInputControl in ShadowPopover](https://github.com/WordPress/gutenberg/assets/1682452/d0d6d614-e98d-42b8-bd3c-81b3df614261)

> [!IMPORTANT]
> So far, as a user I learned that this icon button purpose is to 'switch' a control to an alternative version.

As a user, it took me a while to 'learn' what this icon button is about. Now that I know what it is about, I would expect this icon to _always be used consistently_ and always have the same purpose.

Instead, this is not the case.

4
The fourth occurrence of this icon usage is for the DataViews 'View options` button. Screenshot:

![DataViews View actions](https://github.com/WordPress/gutenberg/assets/1682452/0fb2e98e-1880-4466-8ec8-7ea0c41500eb)

It opens a dropdown menu with a multitude of options for the data views layout, filtering, items to show, items per page. In this context, this icon button meaning and purpose are entirely different.

> [!IMPORTANT]
> As a user, I previously learned this icon means 'switch this control to an alternative version'. But now, suddenly, it means something else. 

IMHO, this inconsistent usage doesn't help in building a cohesive design language. Also, as a user, the cognitive load is now doubled because I have to remember that this icon has different meanings in different contexts.

5
The fifth occurrence of this icon is in the Query loop block toolbar. Screenshot:

![Query block Display settings](https://github.com/WordPress/gutenberg/assets/1682452/7909cb52-5cac-48f2-9fdd-a94cdaeb57ad)

It gives access to a popover where users can set the items per page, the offset, and the max pages to show for the Query.

Again, the icon meaning and purpose in this context is entirely different and inconsistent. As a user, I have now to remember that this icon means something entirely different in three different contexts.

## Conclusion
The 'settings' icon is a good example of how the actual icons _usage_ can be problematic when it's inconsistent. Ideally, icons that need to be 'learned' should have only _one, specific, consistent_ usage in an user interface. There is no point in forcing users to _learn_ what an icon means when the icon meaning changes depending on the context.

Overall, I'd like to see a more strict, cohesive usage of icons throughout the editor, especially for icons that users have to 'learn'. I'm not sure the current situation helps much with building a good user experience. 

To me, the cases 4) and 5) are less than ideal. Those controls should be changed to use another icon. The `settings` icon should be reserved to the 'switch' button for the controls alternative version.

The editor is already complicated. I'd think design should strive to _simplify wherever possible_ and have UI conventions that are consistent to reduce cognitive load as much as possible.

### Step-by-step reproduction instructions

N/A
Go through the various UIs illustrated in this issue description.
Observe the inconsistent icon usage.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by richtabor** [comment]
> The fourth occurrence of this icon usage is for the DataViews 'View options` button. Screenshot:

Yes, this should be changed. Perhaps one of the layout grid icons, or the filter icon we're using elsewhere.

**Comment by richtabor** [comment]
> The fifth occurrence of this icon is in the Query loop block toolbar. Screenshot:

These settings are better suited for the Inspector anyhow; I'd move them out of the toolbar completely. We have most query controls in the Inspector, apart from these; it's quite disorienting.

**Comment by afercia** [comment]
Specifically to the Query block, worth reminding the 'Display settings' button in the block toolbar is conditional.
When the Query block _does_ 'Inherit query from template', there's no Display settings button, of course.
It only shows when the Query block _does not_ inherit. See screenshots below.
In this case, I'm not sure I understand why the Items per page, Offset, and Max page(s) settings are separate from 'Order by' and other settings.

I'd agree all these controls should show up in the same place. Moving all of them to the Inspector could make the Inspetor too crowded though but I'd think it would be a good first step to make the Query block usage less disorienting.

Further improvements could be considered in a follow-up. 

![Screenshot 2024-07-01 at 09 37 55](https://github.com/WordPress/gutenberg/assets/1682452/dd39e07b-c14a-456a-ae51-1d425f32061f)

**Comment by afercia** [comment]
Re: the Query block settings, it appears there's an bug. All these `NumberControl` are wrapped within a `BaseControl` component, not sure why:

![numbercontrols](https://github.com/WordPress/gutenberg/assets/1682452/f68d23ca-637c-4988-8c89-24cf0aa5ef2e)

The `help` description of the 'Max page to show' [is set to the wrapping BaseControl](https://github.com/WordPress/gutenberg/blob/214f46655ef698b717591c2d2b05eb7543e1c3c0/packages/block-library/src/query/edit/query-toolbar.js#L94-L116). As such, the description isn't correctly associated via `aria-describedby` to the `NumberControl`.

**Comment by mtias** [comment]
Indeed, this icon has spread too much for related but distinct uses. It was initially introduced for the extended color controls—it's a jump into more advanced, custom, or granular controls. For more general "settings" there is the cog icon to be used.

**Comment by afercia** [comment]
For the DataVIews, I'd lean towards the `cog` icon as well. Not sure we can use one of the layout grid icons, or the filter icon. The View Options menu contains heterogeneous settings that aren't just about layout or filter.

Looking at [the material design icons](https://m3.material.io/styles/icons/overview) for inspiration, something like the 'Dispkay settings' icon could work. In the meantime, I'd use the more generic concept of 'Settings' represented by the cog.

Comparison with some of the material design icons:

![Screenshot 2024-07-01 at 13 02 14](https://github.com/WordPress/gutenberg/assets/1682452/25d951f2-62be-497b-af8f-1e995473c947)

**Comment by afercia** [comment]
@ntsekouras should the 'Items per page', 'Offset', and 'Max pages to show' settings be compatible with `allowedControls` introduced in https://github.com/WordPress/gutenberg/pull/43632? In that case, I'd create a new issue.

**Comment by ntsekouras** [comment]
>@ntsekouras should the 'Items per page', 'Offset', and 'Max pages to show' settings be compatible with allowedControls introduced in https://github.com/WordPress/gutenberg/pull/43632? In that case, I'd create a new issue.

I'd say not for now as they are quite basic important controls and a bit of different from the other controls (part of `allowedControls`). Having said that I don't have a strong opinion on this and eventually I think the direction is to use the new API to render the controls for each attribute, which is going to be more granular and available for all blocks.


Besides the above, I can open a PR to move the 'settings' controls in the inspector.

**Comment by afercia** [comment]
Thanks @ntsekouras I totally defer the `allowedControls` / new API thing to your experties.

> I can open a PR to move the 'settings' controls in the inspector.

I already tried that in this PR: https://github.com/WordPress/gutenberg/pull/63020 I'd appreciate a review, if you have a chance.
